pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -152,6 +158,8 @@ func (m *mergeQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]
 // queriers. It also adds the `idLabelName` and if present in the original
 // results the original `idLabelName`.
 func (m *mergeQuerier) LabelNames() ([]string, storage.Warnings, error) {
+	log, _ := spanlogger.New(m.ctx, "mergeQuerier.LabelNames")
+	defer log.Span.Finish()
 	labelNames, warnings, err := m.mergeDistinctStringSlice(func(ctx context.Context, q storage.Querier) ([]string, storage.Warnings, error) {
```

Comment:
This is interesting because the `mergeDistinctStringSlice` uses `m.ctx` internally. We should pass the context to this function to make it use the right one.

Reply:
But just noticed that it wouldn't make much difference anyways.

Reply:
yes, it won't make much difference because the main pain point that queriers use context passed during creation. but anyway, nice catch. i fix it now to not fix it in future ;)